### ISSUE TITLE
Collection throws KeyError in listing_summary.pt related to thumbnail scales

### ISSUE THREAD

**Issue by fulv** [plone/Products.CMFPlone#1361]
**Plone 5.0.2**

I have a News folder full of news items, each having a lead image.  For some reason, at some point, the aggregator collection (which is pretty much the OOTB one) started throwing the traceback below.

I have identified the place in `plone.scale.storage.AnnotationStorage._cleanup` which is at the root of the problem, and I'm submitting a PR with a fix.

Unfortunately, I don't know ultimately why this condition occurred, and once I patched it locally, I was no longer able to reproduce the problem, because now the annotation has been changed.

However, I'm pretty sure that the bug is real.

```
2016-02-03 14:46:38 ERROR Zope.SiteErrorLog 1454539598.160.36654983718 http://localhost:8081/Plone/news/aggregator/summary_view
Traceback (innermost last):
  Module ZPublisher.Publish, line 138, in publish
  Module ZPublisher.mapply, line 77, in mapply
  Module ZPublisher.Publish, line 48, in call_object
  Module Products.Five.browser.metaconfigure, line 485, in __call__
  Module Products.Five.browser.pagetemplatefile, line 125, in __call__
  Module Products.Five.browser.pagetemplatefile, line 59, in __call__
  Module zope.pagetemplate.pagetemplate, line 132, in pt_render
  Module five.pt.engine, line 93, in __call__
  Module z3c.pt.pagetemplate, line 163, in render
  Module chameleon.zpt.template, line 258, in render
  Module chameleon.template, line 191, in render
  Module chameleon.template, line 171, in render
  Module d9ed701f590bdb145cdc6c40ae68397a.py, line 535, in render
  Module ea4c2289f3b7aaa41b56eb86e99c46fe.py, line 1223, in render_master
  Module ea4c2289f3b7aaa41b56eb86e99c46fe.py, line 420, in render_content
  Module d9ed701f590bdb145cdc6c40ae68397a.py, line 523, in __fill_content_core
  Module ad2b5f3551c575d7e225e74f5ef176e9.py, line 439, in render_content_core
  Module ad2b5f3551c575d7e225e74f5ef176e9.py, line 297, in render_listing
  Module d9ed701f590bdb145cdc6c40ae68397a.py, line 511, in __fill_entries
  Module ad2b5f3551c575d7e225e74f5ef176e9.py, line 1494, in render_entries
  Module d9ed701f590bdb145cdc6c40ae68397a.py, line 345, in __fill_entry
  Module plone.namedfile.scaling, line 322, in scale
  Module plone.scale.storage, line 166, in scale
  Module plone.scale.storage, line 187, in _cleanup
  Module persistent.mapping, line 59, in __delitem__
  Module UserDict, line 25, in __delitem__
KeyError: (('direction', 'thumbnail'), ('fieldname', 'image'), ('height', 128), ('width', 128))

 - Expression: "python:scales.scale('image', 'thumb')"
 - Filename:   ... ne/app/contenttypes/browser/templates/listing_summary.pt
 - Location:   (line 34: col 36)
 - Source:     scale python:scales.scale('image', 'thumb')"
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Arguments:  repeat: {...} (0)
               template: <ViewPageTemplateFile - at 0x7fe841c4e090>
               views: <ViewMapper - at 0x7fe8406a6090>
               modules: <instance - at 0x7fe84ecedef0>
               args: <tuple - at 0x7fe8560b7050>
               here: <ImplicitAcquisitionWrapper aggregator at 0x7fe842834cd0>
               user: <ImplicitAcquisitionWrapper - at 0x7fe842301140>
               nothing: <NoneType - at 0x91a870>
               container: <ImplicitAcquisitionWrapper aggregator at 0x7fe842834cd0>
               request: <instance - at 0x7fe840bbce18>
               wrapped_repeat: <SafeMapping - at 0x7fe8414a3208>
               traverse_subpath: <list - at 0x7fe84150f1b8>
               default: <object - at 0x7fe855fd3500>
               loop: {...} (1)
               context: <ImplicitAcquisitionWrapper aggregator at 0x7fe842834cd0>
               view: <SimpleViewClass from /usr/local/plone-5.0/buildout-cache/eggs/plone.app.contenttypes-1.2.9-py2.7.egg/plone/app/contenttypes/browser/templates/listing_summary.pt summary_view at 0x7fe84072f110>
               translate: <function translate at 0x7fe8408205f0>
               root: <ImplicitAcquisitionWrapper Zope at 0x7fe8423157d0>
               options: {...} (0)
               target_language: <NoneType - at 0x91a870>
```

**Comment by fulv** [comment]
https://github.com/plone/plone.scale/pull/8

**Comment by mauritsvanrees** [comment]
Your fix was released a while ago in plone.scale 1.4.1, available in Plone 4.3.8 and 5.0.3.
